### PR TITLE
Add application unit test

### PR DESCRIPTION
**Pull request by zigen**
this PR added the unit test for the Application and also introduced `TestSimulation`.
`TestSimulation` allows us to write the test which checks the packet communicating with other modules through the `TestGate`

see https://github.com/sfc-aqua/quisp/pull/215/files?short_path=0d79950#diff-0d799505ba0fecf382a6f51dd2721a43602cb51b285fccd7bea90958bb4bb42c

### CODE REVIEW

File: quisp/test_utils/StaticEnv.h [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+namespace quisp_test {
+namespace env {
+
+using namespace omnetpp;
+
+/**
+ * \brief cEnvir implementation for unit testing based on OMNeT++'s StaticEnv.
+ * \see https://github.com/omnetpp/omnetpp/blob/308b3b2b4c060b165911237a9bd9470ea010d34c/src/sim/csimulation.cc#L728
+ */
+class StaticEnv : public omnetpp::cEnvir {
```
Comment (Chibikuri) [inline quisp/test_utils/StaticEnv.h:15]: This class supports some environmental functions of omnet, right?

Comment (zigen) [reply quisp/test_utils/StaticEnv.h:15]: yes, it inherits cEnvir. 
and I added some utility methods for testing

File: quisp/test_utils/ModuleType.h [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  cModule *createModuleObject() override { return nullptr; };
+  void addParametersAndGatesTo(cModule *module) override{};
+  void applyPatternAssignments(cComponent *component) override{};
+  void setupGateVectors(cModule *module) override{};
+  void buildInside(cModule *module) override{};
+
+  // methods redefined from cComponentType
+  cProperties *getProperties() const override { return props; };
+  cProperties *getParamProperties(const char *paramName) const override { return nullptr; };
```
Comment (Chibikuri) [inline quisp/test_utils/ModuleType.h:32]: Now here most of the functions return nullptr. When we actually use this, we have to properly implement the functions at .cc ?

Comment (zigen) [reply quisp/test_utils/ModuleType.h:32]: yeah if our components need the return value of this method, somehow we need to implement it.

**Review by Chibikuri — APPROVED** [review]
I tested on my environment and it seems working properly. 
This looks very useful.

**Pull request merged**